dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/BitstreamFormatRestRepository.java

```diff
@@ -49,18 +49,20 @@ public BitstreamFormatRest findOne(Context context, Integer id) {
 		if (bit == null) {
 			return null;
 		}
-		return converter.fromModel(bit);
+		return converter.fromModel(utils.applyProjection(bit, projection));
```

Comment:
this mean that the converter can receive a "limited" instance of the model class where some relations are set to null to avoid lazy loading. We want also to be able to expand the response including embedded resources such as subcommunities or collections in a CommunityRest response. To do that we could rely on the HAL wrapper constructor that is already prearranged to receive the list of relations to explode invoking external LinkRestRepository (see https://wiki.duraspace.org/display/DSPACE/DSpace+7+REST%3A+Coding+DSpace+Objects#DSpace7REST:CodingDSpaceObjects-LinkedRepositoryObject(TODO-proposal) and https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/model/hateoas/DSpaceResource.java#L71

Reply:
I must admit that I do not yet fully understand that part of the code. Do you mean that projections should also be able to filter the list of returned links? Since the results are retrieved through a dedicated `LinkRestRepository`, I don't there is a conflict with the projections here.

Reply:
I'm saying that we need to combine the two approach. Your proposed implementation only take care to limit the amount of retrieved / returned information. Indeed, projection should be able also to force the embedding of linked entities that are not part of the Rest object and therefore not included in the "default" serialization. This could be done "translating" the projection parameters in the list of relations to include providing such list to the HAL wrapper, see https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/model/hateoas/BrowseIndexResource.java#L24 for an example of hardcoded "projection" that we should solve as part of this implementation